### PR TITLE
[docs] Cilium Gateway API implementation

### PR DESCRIPTION
**Pull request by lizrice**
Add Cilium to list of Gateway API implementations

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
Adds Cilium to list of Gateway API implementations 

**Which issue(s) this PR fixes**:

N/A 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: site-src/implementations.md [hunk 1]
```diff
@@ -61,6 +63,22 @@ APISIX currently supports Gateway API `v1alpha2` version of the specification fo
...
+
+[Cilium][cilium] is an eBPF-based networking, observability and security
+solution for Kubernetes and other networking environments. It includes [Cilium
+Service Mesh][cilium-service-mesh], a highly efficient mesh data plane that can
+be run in [sidecarless mode][cilium-sidecarless] to dramatically improve
+performance, and avoid the operational complexity of sidecars. Cilium also
+supports the sidecar proxy model, offering choice to users. Cilium is working on
+a Gateway API implementation.
+
+Cilium is open source and is a CNCF incubation project.
```
Comment (markmc) [inline site-src/implementations.md:76]: Implementations are starting to add contact information for issues relating to a specific implementation. Care to do the same for Cilium?

See #1278, #1277, and #1224

Thanks!

File: site-src/implementations.md [hunk 2]
```diff
@@ -61,6 +63,22 @@ APISIX currently supports Gateway API `v1alpha2` version of the specification fo
...
 
+### Cilium
+
+[Cilium][cilium] is an eBPF-based networking, observability and security
+solution for Kubernetes and other networking environments. It includes [Cilium
+Service Mesh][cilium-service-mesh], a highly efficient mesh data plane that can
+be run in [sidecarless mode][cilium-sidecarless] to dramatically improve
+performance, and avoid the operational complexity of sidecars. Cilium also
+supports the sidecar proxy model, offering choice to users. Cilium is working on
+a Gateway API implementation.
```
Comment (markmc) [inline site-src/implementations.md:74]: A link to any info specific to Gateway API would be great - a work-in-progress PR, or issue tracking the implementation, etc. I couldn't find anything with a quick search

Comment (shaneutt) [reply site-src/implementations.md:74]: I agree, ideally we should have some kind of link here back to Cilium's implementation that viewers can follow.

Comment (lizrice) [reply site-src/implementations.md]: Added

File: site-src/implementations.md (hunk 1 above)
Comment (lizrice) [reply site-src/implementations.md]: Good idea, added

**Comment by markmc** [comment]
Excellent, thank you

/lgtm

**Review by shaneutt — APPROVED** [review]
/lgtm

**Pull request merged**